po-box-key.js

```diff
@@ -15,6 +15,16 @@ const length = 32
 const salt = SHA256(SALT)
 
 function poBoxKey (x_dh_secret, x_dh_public, x_id, y_dh_public, y_id) {
+  if (isKeyDM(x_dh_public)) {
+    if (!isKeyPOBox(y_dh_public)) throw new Error('exactly 1 poBox expected')
+    if (!isKeyDM(x_dh_public)) throw new Error('x_dh_public/secret must be same BFE type-format')
+  } // eslint-disable-line
+  else if (isKeyPOBox(x_dh_public)) {
+    if (!isKeyDM(y_dh_public)) throw new Error('exactly 1 poBox expected')
+    if (!isKeyPOBox(x_dh_public)) throw new Error('x_dh_public/secret must be same BFE type-format')
+  } // eslint-disable-line
+  else throw new Error('wrong BFE keys provided')
```

Comment:
While solving the above problem, I realised there were no checks on the type/ format of keys being passed in (which is fatal).
So I added tests and checks here.

Reply:
This turned up a bug in the test vectors for POBox encryption, which is now patched and merged, and the test vector now passes these tests (where before it failed!)